evo1/seq/overworld.py

```diff
@@ -74,56 +58,68 @@ def _should_manip(self) -> bool:
             # Else, go for the encounter that gives the most gli
             return manip_gli > next_gli
 
+    class _MANIP_FSM(Enum):
+        CAN_MANIP = auto()
+        STARTED_MANIP = auto()
+        MANIP_DONE = auto()
+
     # Returning true means we seize control instead of moving on
     def do_encounter_manip(self) -> bool:
-        # Check if we've already picked up the chest
-        if not self.can_manip:
-            return False
+        # Update the next expected encounter
+        self.calc_next_encounter(small_sword=True)
         # Check if we are done farming
         if self._farm_done():
             return False
-        player = get_zelda_memory().player
-        dist_to_chest = dist(player.pos, self._CHEST_LOCATION)
 
-        # Check if we are outside range of the chest for next enc
```

Comment:
This logic was pretty fragile and difficult to follow, so it's replaced with a finite state machine.